src/api/gnosisProtocol/legacy/types.ts

```diff
@@ -5,11 +5,11 @@ import { FeeInformation, PriceInformation } from 'types'
 
 import { GpPriceStrategy } from 'legacy/state/gas/atoms'
 
-type FeeQuoteParams = Pick<EnrichedOrder, 'sellToken' | 'buyToken' | 'kind'> & {
+interface FeeQuoteParams extends Pick<EnrichedOrder, 'sellToken' | 'buyToken' | 'kind'> {
```

Comment:
@alfetopito for some reason, IDEA doesn't highlight implemented fields when I use `type` instead of `interface`. Do you have the same or it's my local issue?

With interfaces:

<img width="869" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/d9cbb1cc-76c4-4467-bece-d404bacad9f5">

With types:

<img width="886" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/b4a90aeb-735b-4db1-8144-e31268bda756">

Reply:
Not at all.

I changed to types based on this PR:

```bash
$ git diff
diff --git a/src/api/gnosisProtocol/legacy/types.ts b/src/api/gnosisProtocol/legacy/types.ts
index 59f74cd80..93c56b314 100644
--- a/src/api/gnosisProtocol/legacy/types.ts
+++ b/src/api/gnosisProtocol/legacy/types.ts
@@ -1,11 +1,10 @@
-import { SupportedChainId as ChainId, PriceQuality } from '@cowprotocol/cow-sdk'
-import { EnrichedOrder } from '@cowprotocol/cow-sdk'
+import { EnrichedOrder, PriceQuality, SupportedChainId as ChainId } from '@cowprotocol/cow-sdk'

 import { FeeInformation, PriceInformation } from 'types'

 import { GpPriceStrategy } from 'legacy/state/gas/atoms'

-interface FeeQuoteParams extends Pick<EnrichedOrder, 'sellToken' | 'buyToken' | 'kind'> {
+type FeeQuoteParams = Pick<EnrichedOrder, 'sellToken' | 'buyToken' | 'kind'> & {
   amount: string
   userAddress?: string | null
   receiver?: string | null
@@ -31,7 +30,7 @@ export class LegacyPriceQuoteError extends Error {
   }
 }

-export interface LegacyFeeQuoteParams extends FeeQuoteParams {
+export type LegacyFeeQuoteParams = FeeQuoteParams & {
   fromDecimals?: number
   toDecimals?: number
   chainId: ChainId
```

No issues 
![image](https://github.com/cowprotocol/cowswap/assets/43217/1e8bade9-b28c-4ee6-9a53-7fc887af9c9c)

Reply:
But do you see inheritance markers on the side-field?

<img width="44" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/a3f28ea5-7365-4f8e-acf8-66e59c266ea8">

Reply:
No, that is not shown 
![image](https://github.com/cowprotocol/cowswap/assets/43217/00a4ce52-8daf-4a34-b763-179ab29bc4af)

I guess if you care about that, it's one good reason for using interfaces

Reply:
BTW, only now I realized that was your question in the first place 🤦 
I saw the squigly red lines and assumed you meant a type error or something.
Sorry about that.